### ISSUE TITLE
don't load unused built-in extensions

### ISSUE THREAD

**Issue by bughit** [brave/brave-browser#6972]
## Description 
In addition to rewards (#3436), the following extensions/processes should not be loaded when their functionality is not activated.

- Utility: Bat Ledger Service (nothing related to BAT has been activated)
- Utility: Bat Ads Service (nothing related to BAT has been activated)
- Extension: Brave Sync (sync has not been activated)

## Brave version (brave://version info)

```
Brave | 1.0.0 Chromium: 78.0.3904.97 (Official Build) (64-bit)
Revision | 021b9028c246d820be17a10e5b393ee90f41375e-refs/branch-heads/3904@{#859}
OS | Windows 7 Service Pack 1 (Build 7601.24535)
```

**Comment by bsclifton** [comment]
Related: https://github.com/brave/brave-browser/issues/5429

**Comment by NejcZdovc** [comment]
@bsclifton yup this is dupe of https://github.com/brave/brave-browser/issues/5429, so closing this one. Will add +1 to it.

**Comment by bughit** [comment]
@NejcZdovc The rewards issue has already been fixed in #3436.  #5429 is also specific to rewards ("Rewards extension background pages should not be running when Rewards is disabled") so should be closed as fixed/dup of #3436. This one should remain open.

**Comment by NejcZdovc** [comment]
#5429 is about background processes (like you listed), where #3436 is about extension being disabled, they are different

**Comment by bughit** [comment]
@NejcZdovc 

no #3436 is about the Rewards extension process, "Rewards extension process needs to be terminated when disabled" and #5429 is about the exact same thing "Rewards extension background pages should not be running when Rewards is disabled"

they are identical and are both already fixed. This issue is about other extensions/processes.

The previous incarnation of this issue (#4612) was already closed as a dup of #3436 and never addressed. And you are doing the same thing with this one. There's a good chance that #5429 will also be closed because at face value, it's already resolved.

It's this issue that should be open.

**Comment by bsclifton** [comment]
@bughit I believe the extra processes you see (which should be owned by the Chromium utility process) should be covered by https://github.com/brave/brave-browser/issues/5429, given what @NejcZdovc captured

@NejcZdovc can you update the issue to include the other utility processes?

**Comment by bughit** [comment]
> I believe the extra processes ... should be covered by #5429

I already pointed out that #3436 was also supposed to cover extra processes (#4612 was closed as a dup of #3436) but didn't. No need to believe when you have evidence.

There are already two perfectly good bugs that explicitly and precisely refer to the extra processes issue (#4612 and #6972) and yet they keep getting closed in favor of issues whose scope is only Rewards.
